libsignal-protocol/src/lib.rs

```diff
@@ -137,3 +137,9 @@ pub trait Deserializable: Sized {
     /// Parse the provided data in the protobuf format.
     fn deserialize(ctx: &Context, data: &[u8]) -> Result<Self, Error>;
 }
+
+/// A helper trait for something that can be encoded to base64
+pub trait Base64 {
```

Comment:
Not sure whether this is really necessary.  I'd be okay with having `to_base64` as a method on both Public and Private key. Are you going to program against a `Base64` interface?

Reply:
Yes, in the `signal-bot-rs` client I just have something to serialize those with `serde` and I don't want to duplicate code. This is also a one-line trait, so I don't see any downside.

Reply:
Then still, shouldn't it be `to_base64`?

I would probably impl Serialize on both types, and where you actually do the serialization (with the types as fields), use one of the [`serde(with)` attributes](https://serde.rs/field-attrs.html#serialize_with), as exists in e.g. [service's `utils` module](https://github.com/Michael-F-Bryan/libsignal-service-rs/blob/master/libsignal-service/src/utils.rs).  After all, base64 encoding is a serialization detail, not a property of the struct IMO.